rs/registry/canister/src/mutations/node_management/do_replace_operator.rs

```diff
@@ -0,0 +1,628 @@
+            .into_iter()
+            .map(|(node_id, node_record)| {
+                let node_key = make_node_record_key(node_id);
+                let updated_node_record = NodeRecord {
+                    node_operator_id: new_operator_id.as_slice().to_vec(),
+                    ..node_record
+                };
+                update(node_key, updated_node_record.encode_to_vec())
+            })
+            .collect();
+
+        // Nothing should be done.
+        if valid_mutations.is_empty() {
+            return Ok(());
+        }
+
+        let node_mutations = valid_mutations.len() as u64;
+
+        if node_mutations > new_operator_record.node_allowance {
+            return Err(format!("{}do_replace_operator: New operator cannot accept {} nodes due to remaining allowance {}", LOG_PREFIX, node_mutations, new_operator_record.node_allowance));
+        }
+
+        // Decrement new operator allowance.
+        let new_node_operator_key = make_node_operator_record_key(new_operator_id);
+        let updated_new_operator_record = NodeOperatorRecord {
+            node_allowance: new_operator_record.node_allowance - node_mutations,
+            ..new_operator_record.clone()
+        };
+        valid_mutations.push(update(
+            new_node_operator_key,
+            updated_new_operator_record.encode_to_vec(),
+        ));
+
+        // Increment old operator allowance.
+        let old_node_operator_key = make_node_operator_record_key(old_operator_id);
+        let updated_old_operator_record = NodeOperatorRecord {
+            node_allowance: old_operator_record.node_allowance + node_mutations,
+            ..old_operator_record.clone()
+        };
+        valid_mutations.push(update(
+            old_node_operator_key,
+            updated_old_operator_record.encode_to_vec(),
+        ));
+
+        self.maybe_apply_mutation_internal(valid_mutations);
+
+        println!(
+            "{}do_replace_operator: Finished executing payload: {:?}",
+            LOG_PREFIX, payload
+        );
+
+        Ok(())
+    }
+
+    /// Return the set of node records that belong to `old_operator_id`.
+    ///
+    /// If the node record is linked to the `new_operator_id`, it will
+    /// be filtered from the results, meaning that not all nodes from
+    /// `provided_node_ids` have to be returned.
+    fn maybe_fetch_nodes_to_update(
+        &self,
+        provided_node_ids: &Vec<NodeId>,
+        new_operator_id: &PrincipalId,
+        old_operator_id: &PrincipalId,
+    ) -> Result<BTreeMap<NodeId, NodeRecord>, String> {
+        let mut node_records = BTreeMap::new();
+
+        for node_id in provided_node_ids {
+            let node_record = self.get_node(*node_id).ok_or_else(|| {
+                format!(
+                    "{}do_replace_operator: Node not found: {}",
+                    LOG_PREFIX, node_id
+                )
+            })?;
+
+            if node_record.node_operator_id.eq(new_operator_id.as_slice()) {
```

Comment:
Why `eq` instead of `==`? I think the latter is how it's usually done, and unless there is some special reason that I'm not seeing, the usual way should be used.